File: ios/Classes/Wrapper/Mappers/MapToProductMapper.swift
//
//  Created by Emarsys on 2021. 10. 21..
//

import EmarsysSDK

class MapToProductMapper: Mappable {

    typealias Input = [String: Any]?
    typealias Output = Product?

    func map(_ input: [String: Any]?) -> Product? {
        guard input != nil else {
            return nil
        }
        return Product(
                productId: input!["productId"] as! String,
                title: input!["title"] as! String,
                linkUrlString: input!["linkUrlString"] as! String,
                feature: input!["feature"] as! String,
                cohort: input!["cohort"] as! String,
                customFields: input!["customFields"] as! [String: Any],
                imageUrlString: input!["imageUrlString"] as! String?,
                zoomImageUrlString: input!["zoomImageUrlString"] as! String?,
                categoryPath: input!["categoryPath"] as! String?,
                available: input!["available"] as! Bool?,
                productDescription: input!["productDescription"] as! String?,
                price: input!["price"] as! Double?,
                msrp: input!["msrp"] as! Double?,
                album: input!["album"] as! String?,
                actor: input!["actor"] as! String?,
                artist: input!["artist"] as! String?,
                author: input!["author"] as! String?,
                brand: input!["brand"] as! String?,
                year: input!["year"] as! Int?
        )
    }
}

File: android/src/main/kotlin/com/emarsys/emarsys_sdk/mapper/MapToProductMapper.kt
package com.emarsys.emarsys_sdk.mapper

import com.emarsys.predict.api.model.Product

class MapToProductMapper : Mapper<Map<String, Any>, Product> {

    override fun map(productMap: Map<String, Any>): Product {
        return Product(
            productId = productMap["productId"] as String,
            title = productMap["title"] as String,
            linkUrl = productMap["linkUrlString"] as String,
            feature = productMap["feature"] as String,
            cohort = productMap["cohort"] as String,
            customFields = productMap["customFields"] as Map<String, String?>,
            imageUrlString = productMap["imageUrlString"] as String?,
            zoomImageUrlString = productMap["zoomImageUrlString"] as String?,
            categoryPath = productMap["categoryPath"] as String?,
            available = productMap["available"] as Boolean?,
            productDescription = productMap["productDescription"] as String?,
            price = (productMap["price"] as Double?)?.toFloat(),
            msrp = (productMap["msrp"] as Double?)?.toFloat(),
            album = productMap["album"] as String?,
            actor = productMap["actor"] as String?,
            artist = productMap["artist"] as String?,
            author = productMap["author"] as String?,
            brand = productMap["brand"] as String?,
            year = productMap["year"] as Int?,
        )
    }
}